From 5baaaf928e61373abbde25e964f1c831d3560215 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Tue, 23 Aug 2016 20:48:04 +0200
Subject: [PATCH 0459/1345] clk: mvebu: revert clock support for CP110

commit  f43da38320eccadfeae71a1e01fa11b0f0e62722 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch reverts following commits as for the preparation of
adding the mainline driver

commit b655f34d2b0a ("clk: cp110: added support for cp110 clock sources")

Conflicts:
	arch/arm64/Kconfig.platforms
	arch/arm64/boot/dts/marvell/armada-cp110.dtsi

commit 67ef4234ab29 ("fix: dts: a8040: disable gateclk_1 in CP1")

This patch is part of patch-set which adds
mainline mvebu clock gating support [1/21].

Change-Id: I900dfec98eb80fd97a0a9386b3ab667be3f04964
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32386
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../clock/mvebu-armada-cp110-core-clock.txt        |   32 -----
 .../bindings/clock/mvebu-gated-clock.txt           |   29 +----
 arch/arm64/Kconfig.platforms                       |    1 -
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi    |   18 ++-
 drivers/clk/mvebu/Kconfig                          |    3 -
 drivers/clk/mvebu/Makefile                         |    1 -
 drivers/clk/mvebu/cp110.c                          |  125 --------------------
 7 files changed, 9 insertions(+), 200 deletions(-)
 delete mode 100644 Documentation/devicetree/bindings/clock/mvebu-armada-cp110-core-clock.txt
 delete mode 100644 drivers/clk/mvebu/cp110.c

diff --git a/Documentation/devicetree/bindings/clock/mvebu-armada-cp110-core-clock.txt b/Documentation/devicetree/bindings/clock/mvebu-armada-cp110-core-clock.txt
deleted file mode 100644
index 6418bde..0000000
--- a/Documentation/devicetree/bindings/clock/mvebu-armada-cp110-core-clock.txt
+++ /dev/null
@@ -1,32 +0,0 @@
-* Clock bindings for Marvell MVEBU CP110 Core clocks
-
-The Marvell MVEBU Armada 7K/8K SoCs contain a block called CP110,
-hosting slow and high speed peripherals. This Device Tree
-binding allows to describe the core clocks of the CP110, whose
-frequencies are fixed except for the NAND clock.
-
-Clock consumers must specify the desired clock by having the clock ID
-in its "clocks" phandle cell.
-
-The following is a list of provided IDs and clock names for the core
-Armada AP806 clocks:
-
- 0 = APLL - The PLL clock. Root of the clock tree
- 1 = PPv2 - The Packet processor clock
- 2 = EIP  - The encryption engine clock
- 3 = Core - The clock of all other consumers (except NAND)
- 4 = Nand - The clock of the NAND interface.
-
-Required properties:
-- compatible: must be be one of the following:
-	"marvell,armada-cp110-clock"
-- reg: must be the register address of the NAND clock control register
-- #clock-cells: from common clock binding; shall be set to 1
-
-Example:
-
-	cp110_clk: clk@0x6F8204 {
-		compatible = "marvell,armada-cp110-clock";
-		reg = <0x440700 0x4>;
-		#clock-cells = <1>;
-	};
diff --git a/Documentation/devicetree/bindings/clock/mvebu-gated-clock.txt b/Documentation/devicetree/bindings/clock/mvebu-gated-clock.txt
index 641faaa..f938ef1 100644
--- a/Documentation/devicetree/bindings/clock/mvebu-gated-clock.txt
+++ b/Documentation/devicetree/bindings/clock/mvebu-gated-clock.txt
@@ -1,38 +1,12 @@
 * Gated Clock bindings for Marvell EBU SoCs
 
-Marvell Armada 370/375/380/385/39x/XP, Dove, Kirkwood, and CP-110 allow some
+Marvell Armada 370/375/380/385/39x/XP, Dove and Kirkwood allow some
 peripheral clocks to be gated to save some power. The clock consumer
 should specify the desired clock by having the clock ID in its
 "clocks" phandle cell. The clock ID is directly mapped to the
 corresponding clock gating control bit in HW to ease manual clock
 lookup in datasheet.
 
-The following is a list of provided IDs for Armada CPN110 component:
-ID	Clock		Peripheral
------------------------------------
-0	Audio		AC97 Cntrl
-1	communit	TDM Comm-Unit
-2	nand		Nand Flash Ctrl
-3	pp		Packet Processor
-4	sd		SD/eMMC
-5	mg		MG
-5	mg_core		MG Core
-7	xor1		XOR Engine 1
-8	xor0		XOR Engine 0
-9	gop		Ethernet GOP
-11	pcie_x1_0	PCIe x1, unit 0
-12	pcie_x1_1	PCIe x1, unit 1
-13	pcie_x4		PCIe x4
-14	pcie_core	PCIe Core
-15	sata0		Sata
-16	satausbcore	Sata/USB Core
-18	sd-mmc		SD/MMC unit
-22	usb3h0		USB3 Host #0
-23	USB3h1		USB3 Host #1
-24	USB3d		USB3 device
-25	eip150		EIP-150
-26	eip197		EIP-197
-
 The following is a list of provided IDs for Armada 370:
 ID	Clock	Peripheral
 -----------------------------------
@@ -210,7 +184,6 @@ ID	Clock	Peripheral
 
 Required properties:
 - compatible : shall be one of the following:
-	"marvell,armada-cp110-gating-clock" - for A8K CP110 component clock gating.
 	"marvell,armada-370-gating-clock" - for Armada 370 SoC clock gating
 	"marvell,armada-375-gating-clock" - for Armada 375 SoC clock gating
 	"marvell,armada-380-gating-clock" - for Armada 380/385 SoC clock gating
diff --git a/arch/arm64/Kconfig.platforms b/arch/arm64/Kconfig.platforms
index 3d9d30e..d503a0b 100644
--- a/arch/arm64/Kconfig.platforms
+++ b/arch/arm64/Kconfig.platforms
@@ -118,7 +118,6 @@ config ARCH_MVEBU
 	select OF_GPIO
 	select PINCTRL_ARMADA_AP806
 	select PINCTRL_ARMADA_CP110
-	select ARMADA_CP110_CLK
 	select PINCTRL
 	select PINCTRL_ARMADA_37XX
 	help
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index a170533..a542b77 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -29,14 +29,12 @@ cp110_clk_1: clk@0x440700 {
 	#clock-cells = <1>;
 };
 
-/* Need to debug freeze when the gateclk_1 is enabled
 gateclk_1: clock-gating-control@440220 {
 	compatible = "marvell,armada-cp110-gating-clock";
 	reg = <0x440220 0x4>;
 	clocks = <&cp110_clk_1 3>;
 	#clock-cells = <1>;
 };
-*/
 
 pinctrl@440000 {
 	reg = <0x440000 0x10>;
@@ -46,7 +44,7 @@ sata@540000 {
 	compatible = "marvell,armada-3700-ahci";
 	reg = <0x540000 0x30000>;
 	interrupts = <GIC_SPI 287 IRQ_TYPE_LEVEL_HIGH>;
-	clocks = <&gateclk 15>, <&gateclk 16>;
+	clocks = <&gateclk_1 15>, <&gateclk_1 16>;
 	status = "disabled";
 	port_base = <0x10000>;
 	port_offset = <0x10000>;
@@ -57,7 +55,7 @@ usb3h0_1: usb3@500000 {
 	reg = <0x500000 0x4000>;
 	dma-coherent;
 	interrupts = <GIC_SPI 286 IRQ_TYPE_LEVEL_HIGH>;
-	clocks = <&gateclk 22>, <&gateclk 16>;
+	clocks = <&gateclk_1 22>, <&gateclk_1 16>;
 	status = "disabled";
 };
 
@@ -66,7 +64,7 @@ usb3h1_1: usb3@510000 {
 	reg = <0x510000 0x4000>;
 	dma-coherent;
 	interrupts = <GIC_SPI 285 IRQ_TYPE_LEVEL_HIGH>;
-	clocks = <&gateclk 23>, <&gateclk 16>;
+	clocks = <&gateclk_1 23>, <&gateclk_1 16>;
 	status = "disabled";
 };
 
@@ -121,7 +119,7 @@ nand@720000 {
 	marvell,nand-enable-arbiter;
 	nand-on-flash-bbt;
 	interrupts = <GIC_SPI 307 IRQ_TYPE_LEVEL_HIGH>;
-	clocks = <&gateclk 2>;
+	clocks = <&gateclk_1 2>;
 	nand-ecc-strength = <4>;
 	nand-ecc-step-size = <512>;
 	status = "disabled";
@@ -133,7 +131,7 @@ cpxor0@6a0000 {
 	      <0x6b0000 0x1000>;
 	dma-coherent;
 	msi-parent = <&gic_v2m0>;
-	clocks = <&gateclk 14>, <&gateclk 8>;
+	clocks = <&gateclk_1 14>, <&gateclk_1 8>;
 	status = "disabled";
 };
 
@@ -143,7 +141,7 @@ cpxor1@6c0000 {
 	      <0x6d0000 0x1000>;
 	dma-coherent;
 	msi-parent = <&gic_v2m0>;
-	clocks = <&gateclk 14>, <&gateclk 7>;
+	clocks = <&gateclk_1 14>, <&gateclk_1 7>;
 	status = "disabled";
 };
 
@@ -267,7 +265,7 @@ ppv22@000000 {
 	      <0x441000 0x1000>;  /* RFU-1 Regs */
 	reg-names = "pp", "serdes", "xmib", "led", "smi", "tai", "xsmi", "mg", "mspg", "xpcs",
 		    "fca", "gmac", "xlg", "rfu1";
-	clocks = <&gateclk 3>, <&gateclk 18>, <&gateclk 9>, <&gateclk 6>, <&gateclk 5>;
+	clocks = <&gateclk_1 3>, <&gateclk_1 18>, <&gateclk_1 9>, <&gateclk_1 6>, <&gateclk_1 5>;
 	clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
 	status = "okay";
 	eth0_1: eth0@010000 {
@@ -306,6 +304,6 @@ eip197@800000 {
 	reg = <0x800000 0x200000>;
 	dma-coherent;
 	interrupts = <GIC_SPI 278 IRQ_TYPE_LEVEL_HIGH>;
-	clocks = <&gateclk 26>;
+	clocks = <&gateclk_1 26>;
 	status = "disabled";
 };
diff --git a/drivers/clk/mvebu/Kconfig b/drivers/clk/mvebu/Kconfig
index 57100e0..8d627fc 100644
--- a/drivers/clk/mvebu/Kconfig
+++ b/drivers/clk/mvebu/Kconfig
@@ -65,6 +65,3 @@ config ARMADA_AP806_CORE_CLK
 
 config ARMADA_AP806_RING_CLK
 	bool
-
-config ARMADA_CP110_CLK
-	bool
diff --git a/drivers/clk/mvebu/Makefile b/drivers/clk/mvebu/Makefile
index 734d3d1..1d441e2 100644
--- a/drivers/clk/mvebu/Makefile
+++ b/drivers/clk/mvebu/Makefile
@@ -9,7 +9,6 @@ obj-$(CONFIG_ARMADA_3700_CLK)	+= armada-3700.o
 obj-$(CONFIG_MSYS_CLK)	+= msys.o
 obj-$(CONFIG_ARMADA_AP806_CORE_CLK) += ap806-core.o
 obj-$(CONFIG_ARMADA_AP806_RING_CLK) += ap806-ring.o
-obj-$(CONFIG_ARMADA_CP110_CLK)  += cp110.o
 obj-$(CONFIG_ARMADA_37XX_CLK)	+= armada-37xx-xtal.o
 obj-$(CONFIG_ARMADA_37XX_CLK)	+= armada-37xx-tbg.o
 obj-$(CONFIG_ARMADA_37XX_CLK)	+= armada-37xx-periph.o
diff --git a/drivers/clk/mvebu/cp110.c b/drivers/clk/mvebu/cp110.c
deleted file mode 100644
index 6a5ad0d..0000000
--- a/drivers/clk/mvebu/cp110.c
+++ /dev/null
@@ -1,125 +0,0 @@
-/***************************************************************************
-* Copyright (C) 2015 Marvell International Ltd.
-* ***************************************************************************
-* This program is free software: you can redistribute it and/or modify it
-* under the terms of the GNU General Public License as published by the Free
-* Software Foundation, either version 2 of the License, or any later version.
-*
-* This program is distributed in the hope that it will be useful,
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-* GNU General Public License for more details.
-*
-* You should have received a copy of the GNU General Public License
-* along with this program.  If not, see <http://www.gnu.org/licenses/>.
-* ***************************************************************************
-*/
-
-#include <linux/kernel.h>
-#include <linux/clk-provider.h>
-#include <linux/io.h>
-#include <linux/of.h>
-#include <linux/of_address.h>
-#include <linux/slab.h>
-#include "common.h"
-
-#define CPN110_APLL_CLK_FREQ	(1000000000)
-
-/* NAND clock frequency control bit (250 / 400) */
-#define NF_CLOCK_SEL_400_MASK       (0x1)
-
-enum cp110_clock_id {
-	CP110_APLL_CLK = 0,
-	CP110_PPV2_CLK = 1,
-	CP110_EIP_CLK  = 2,
-	CP110_CORE_CLK = 3,
-	CP110_NAND_CLK = 4,
-	CP110_CLK_CNT
-};
-
-static struct clk *cp110_core_clks[CP110_CLK_CNT];
-
-static struct clk_onecell_data cp110_core_clk_data = {
-	.clks = cp110_core_clks,
-	.clk_num = CP110_CLK_CNT,
-};
-
-static void __init cp110_clk_init(struct device_node *np)
-{
-	void __iomem *base;
-	u32 nand_clk_ctrl;
-
-	base = of_iomap(np, 0);
-	if (WARN_ON(!base))
-		return;
-
-	nand_clk_ctrl = readl(base);
-	iounmap(base);
-
-	/* Register the APLL which is the root of the clk tree */
-	cp110_core_clks[CP110_APLL_CLK] = clk_register_fixed_rate(NULL, "apll-clk",
-			NULL, CLK_IS_ROOT, CPN110_APLL_CLK_FREQ);
-
-	/* PPv2 is APLL/3 */
-	cp110_core_clks[CP110_PPV2_CLK] = clk_register_fixed_factor(NULL, "ppv2-clk",
-			"apll-clk", 0, 1, 3);
-
-	/* EIP clock is APLL/2 */
-	cp110_core_clks[CP110_EIP_CLK] = clk_register_fixed_factor(NULL, "eip-clk",
-			"apll-clk", 0, 1, 2);
-
-	/* Core clock is EIP/2 */
-	cp110_core_clks[CP110_CORE_CLK] = clk_register_fixed_factor(NULL, "core-clk",
-			"eip-clk", 0, 1, 2);
-
-	/* NAND can be either APLL/2.5 or core clock */
-	if (nand_clk_ctrl & NF_CLOCK_SEL_400_MASK)
-		cp110_core_clks[CP110_NAND_CLK] = clk_register_fixed_factor(NULL, "nand-clk",
-				"apll-clk", 0, 2, 5);
-	else
-		cp110_core_clks[CP110_NAND_CLK] = clk_register_fixed_factor(NULL, "nand-clk",
-				"core-clk", 0, 1, 1);
-
-	of_clk_add_provider(np, of_clk_src_onecell_get,
-			    &cp110_core_clk_data);
-
-}
-
-CLK_OF_DECLARE(cp110_clk, "marvell,armada-cp110-clock", cp110_clk_init);
-
-/*
- * Clock Gating Control
- */
-static const struct clk_gating_soc_desc armada_cp110_gating_desc[] __initconst = {
-
-	{"audio", NULL, 0},
-	{"communit", NULL, 1},
-	{"nand", "nand-clk", 2},
-	{"pp", "ppv2-clk", 3},
-	{"sd", NULL, 4},
-	{"mg", NULL, 5},
-	{"mg_core", NULL, 6},
-	{"xor1", NULL, 7},
-	{"xor0", NULL, 8},
-	{"gop", NULL, 9},
-	{"pcie_x1_0", NULL, 11},
-	{"pcie_x1_1", NULL, 12},
-	{"pcie_x4", NULL, 13},
-	{"pcie_core", NULL, 14},
-	{"sata", NULL, 15},
-	{"sata_usb_core", NULL, 16},
-	{"sdmmc_core", NULL, 18},
-	{"usb3h0", NULL, 22},
-	{"usb3h1", NULL, 23},
-	{"usb3d", NULL, 24},
-	{"eip150", "eip-clk", 25},
-	{"eip197", "eip-clk", 26},
-	{ }
-};
-
-static void __init armada_cp110_clk_gating_init(struct device_node *np)
-{
-	mvebu_clk_gating_setup(np, armada_cp110_gating_desc);
-}
-CLK_OF_DECLARE(armada_cp110_clk_gating, "marvell,armada-cp110-gating-clock",
-	       armada_cp110_clk_gating_init);
-- 
1.7.9.5

